src/demo/components/ImageSlider.tsx

```diff
@@ -36,68 +37,93 @@ const ImageSlider: React.FC<ImageSliderProps> = ({ images, linkTexts }) => {
 
   return (
     <>
-    <div className="slider-container">
-      <div
-        className={`arrow left display-none ${currentPreviewIndex === 0 ? "disabled" : ""}`}
-        onClick={slidePrevious}
-      >
-        <img src={LeftArrow} alt="left-arrow" />
-      </div>
-      <div className="slider">
-        <div className="side-image left-side-image">
-          <img
-            src={
-              images[(currentPreviewIndex - 1 + images.length) % images.length]
-            }
-            alt={`Image ${currentPreviewIndex - 1}`}
-          />
-          <p className="side-image-text">
-            {
-              linkTexts[
-                (currentPreviewIndex - 1 + images.length) % images.length
-              ]["text"]
-            }
-          </p>
-        </div>
+      <div className="slider-container" ref={sliderRef}>
         <div
-          className="center-image"
-          onClick={() => navigation(linkTexts[currentPreviewIndex]["url"])}
+          className={`arrow left display-none ${
+            currentPreviewIndex === 0 ? "disabled" : ""
+          }`}
+          onClick={slidePrevious}
         >
-          <img src={imageUrl} alt={`Image ${currentPreviewIndex}`} />
-          <p className="center-image-text ">
-            {linkTexts[currentPreviewIndex]["text"]}
-          </p>
+          <img src={LeftArrow} alt="left-arrow" />
+        </div>
+        <div className="slider web-view">
+          <div className="side-image left-side-image">
+            <img
+              src={
+                images[
+                  (currentPreviewIndex - 1 + images.length) % images.length
+                ]
+              }
+              alt={`Image ${currentPreviewIndex - 1}`}
+            />
+            <p className="side-image-text">
+              {
+                linkTexts[
+                  (currentPreviewIndex - 1 + images.length) % images.length
+                ]["text"]
+              }
+            </p>
+          </div>
+          <div
+            className="center-image"
+            onClick={() => navigation(linkTexts[currentPreviewIndex]["url"])}
+          >
+            <img src={imageUrl} alt={`Image ${currentPreviewIndex}`} />
+            <p className="center-image-text ">
+              {linkTexts[currentPreviewIndex]["text"]}
+            </p>
+          </div>
+          <div className="side-image right-side-image">
+            <img
+              src={images[(currentPreviewIndex + 1) % images.length]}
+              alt={`Image ${currentPreviewIndex + 1}`}
+            />
+            <p className="side-image-text">
+              {linkTexts[(currentPreviewIndex + 1) % images.length]["text"]}
```

Comment:
(currentPreviewIndex + 1) % images.length
(currentPreviewIndex - 1) % images.length

You seem to use this formular several times...

Suggest you make helper functions to do this
e.g. `nextIndex(index): index`